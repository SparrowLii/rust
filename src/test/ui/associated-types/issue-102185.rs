// compile-flags: -Zdeduplicate-diagnostics=yes

fn f<T>(_: impl A<X = T>) {}
                //~^ ERROR the trait bound `T: B` is not satisfied [E0277]
trait A: C<Z = <<Self as A>::X as B>::Y> {
    type X: B;
}

trait B {
    type Y;
}

trait C {
    type Z;
}

fn main() {}
