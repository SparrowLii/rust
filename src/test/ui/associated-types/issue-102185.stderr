error[E0277]: the trait bound `T: B` is not satisfied
  --> $DIR/issue-102185.rs:3:17
   |
LL | fn f<T>(_: impl A<X = T>) {}
   |                 ^^^^^^^^ the trait `B` is not implemented for `T`
   |
help: consider restricting type parameter `T`
   |
LL | fn f<T: B>(_: impl A<X = T>) {}
   |       +++

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
